React InstantSearch/routing-basic/src/App.js

```diff
@@ -25,70 +25,53 @@ const searchStateToUrl = (props, searchState) =>
 
 const urlToSearchState = location => qs.parse(location.search.slice(1));
 
-class App extends Component {
-  state = {
-    searchState: urlToSearchState(this.props.location),
-    lastLocation: this.props.location,
-  };
+export function App(props) {
+  const { location, history } = props;
+  const [searchState, setSearchState] = useState(urlToSearchState(location));
+  const debouncedSetState = useRef(null);
 
-  static getDerivedStateFromProps(props, state) {
-    if (props.location !== state.lastLocation) {
-      return {
-        searchState: urlToSearchState(props.location),
-        lastLocation: props.location,
-      };
-    }
+  function onSearchStateChange(updatedSearchState) {
+    clearTimeout(debouncedSetState.current);
 
-    return null;
-  }
-
-  onSearchStateChange = searchState => {
-    clearTimeout(this.debouncedSetState);
-
-    this.debouncedSetState = setTimeout(() => {
-      this.props.history.push(
-        searchStateToUrl(this.props, searchState),
-        searchState
-      );
+    debouncedSetState.current = setTimeout(() => {
+      history.push(searchStateToUrl(props, updatedSearchState));
```

Comment:
We just use `location` in `searchStateToUrl()`, so to make the example easier for users to modify, I don't think we should forward the whole `props` here (they would end up with other props if their `<App>` accept more props). It comes closer to the signature of `urlToSearchState()` too.

Suggestion:

```suggestion
      history.push(searchStateToUrl(location, updatedSearchState));
```

And then, we can destructure the props in `<App>`, as we're doing in `<Hit>`.

Reply:
Yep it makes sense, addressed in 96b2075.